oak_containers_orchestrator/src/client.rs

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod proto {
+    pub mod oak {
+        pub mod containers {
+            tonic::include_proto!("oak.containers");
+        }
+    }
+}
+
+use anyhow::Context;
+use proto::oak::containers::launcher_client::LauncherClient as GrpcLauncherClient;
+use tokio_vsock::VsockStream;
+use tonic::transport::{Endpoint, Uri};
+use tower::service_fn;
+
+// Virtio VSOCK does not use URIs, hence this URI will never be used.
+// It is defined purely since in order to create a channel, since a URI has to
+// be supplied to create an `Endpoint`.
+static IGNORED_ENDPOINT_URI: &str = "file://[::]:0";
```

Comment:
Weird that declaring an ignored endpoint is necessary. Is this completely ignored? Why did you choose "file://[::]:0" then